### PR TITLE
Add py311 subports for 28 Python modules

### PR DESCRIPTION
**Pull request by markmentovai**
#### Description

Python 3.11 subports are added for the following modules:

 - `py311-crcmod`
 - `py311-mock`
 - `py311-freezegun`
 - `py311-coverage`
 - `py311-toml`
 - `py311-pytest-cov`
 - `py311-six`
 - `py311-tz`
 - `py311-dateutil`
 - `py311-geographiclib`
 - `py311-numpy`
 - `py311-iso8601`
 - `py311-cpuinfo`
 - `py311-pytest-benchmark`
 - `py311-setuptools_scm_git_archive`
 - `py311-pytest-subtests`
 - `py311-execnet`
 - `py311-pytest-forked`
 - `py311-pytest-xdist`
 - `py311-cryptography`
 - `py311-jwt`
 - `py311-blinker`
 - `py311-oauthlib`
 - `py311-pbr`
 - `py311-requests-mock`
 - `py311-requests-oauthlib`
 - `py311-svgelements`
 - `py311-yapf`

I’m sending this as a single pull request with independent commits rather than a series of 28 pull requests because many of the new py311 subports depend on others in this group. The commit sequence ensures that no new subport will fail for a missing dependency. Dependencies within this pull request are annotated in commit messages.

In most instances, this simply adds `py311` to `python.versions`. These special cases occurred because the PEP517 build is now the default for Python 3.11 subports:

 - When using the PEP517 build, `py-crcmod` cannot tolerate the `-f` build argument, so this is made conditional on `python.pep517`.
 - When using the PEP517 build, `py-numpy` failed because the bundled versioneer.py in numpy 1.23.5 hasn’t caught up with https://github.com/python-versioneer/python-versioneer/commit/c56a7e4559788b16153e87c927b086ba78f89ef0. This was recently brought into `py-numpy` as part of https://github.com/numpy/numpy/commit/b6d94fab5a7ef8897fe06d0b7a161bc8080fff26, not yet in any numpy release. Rather than patching versioneer.py, this conditionally (based on `python.pep517`) adds the `--skip-dependency-check` build argument, which avoids the problem. This is durable, as even if versioneer.py were patched so that the dependency check could be enabled, the check would insist on rather old pinned versions such as `wheel==0.37.0` (MacPorts has 0.38.4) and `setuptools==59.2.0` (MacPorts has 65.6.0). That precise `setuptools` requirement was set [rather intentionally](https://github.com/numpy/numpy/issues/22623#issuecomment-1320890687). Note that `--skip-dependency-check` restores the same behavior as is used in the non-PEP517 build.
 - A similar situation exists for `py-pytest-forked`, whose `pyproject.toml` specifies `setuptools ~= 41.4`, `setuptools_scm ~= 3.3` (MacPorts has 7.0.5), and `wheel ~= 0.33.6`.

I ran all tests. Many of these ports do not have tests. Most that do pass all tests. Some ports have tests that were already failing for their py310 subports, and are still failing for their new py311 subports. There are no ports with new test failures. Failures, which appeared harmless, were discovered in py310 and py311 subports for:

 - `py-dateutil`
 - `py-cryptography`
 - `py-freezegun`
 - `py-mock`
 - `py-pytest-benchmark`

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 13.0.1 22A400 arm64
Xcode 14.1 14B47b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Review by lpsinger — APPROVED** [review]

**Comment by catap** [comment]
Thanks for this huge work!

File: python/py-six/Portfile
Comment (michaelld) [inline python/py-six/Portfile:29]: Duplicate `310` here

Comment (markmentovai) [reply python/py-six/Portfile]: Thanks, fixed.

**Comment by markmentovai** [comment]
The build failures for `py27-iso8601`, `py35-iso8601`, and `py36-iso8601` were not caused by this change, but are existing problems with those subports due to an unspecified dependency. #16781 fixes that problem.

**Comment by tobypeterson** [comment]
> The build failures for `py27-iso8601`, `py35-iso8601`, and `py36-iso8601` were not caused by this change, but are existing problems with those subports due to an unspecified dependency. #16781 fixes that problem.

Those python versions are all EOL, and nothing appears to depend on those subports. Better to remove them.

**Comment by markmentovai** [comment]
> > The build failures for `py27-iso8601`, `py35-iso8601`, and `py36-iso8601` were not caused by this change, but are existing problems with those subports due to an unspecified dependency. #16781 fixes that problem.
> 
> Those python versions are all EOL, and nothing appears to depend on those subports. Better to remove them.

I updated #16781 to remove those broken subports instead of fixing them.

**Review by tobypeterson — APPROVED** [review]

**Review by stromnov — APPROVED** [review]

**Review by reneeotten — APPROVED** [review]

**Comment by markmentovai** [comment]
Rebased because the following py311 subports were added earlier today:

 - `py-six` (2b291ef5aab4)
 - `py-tz` (e7a23905264b)
 - `py-dateutil` (37f63c4746d1)
 - `py-setuptools_scm_git_archive` (ac6057df89df)
 - `py-cryptography` (a45f5af458b3)

**Comment by markmentovai** [comment]
This should be ready to merge now, as it’s been over 72 hours since it’s been proposed. 5 reviewers have approved, another 3 have given a thumbs-up, and only 1 reviewer hasn’t commented at all; all of the ports that reviewer is the maintainer of are also listed as openmaintainer. Since this is uncontentious and will unblock further Python 3.11 work, it should be merged.

**Comment by tobypeterson** [comment]
Done. Thanks!

**Comment by harens** [comment]
https://github.com/macports/macports-ports/blob/e312a74f9fa930ace01664cca38937a46becabb8/python/py-numpy/Portfile#L133-L135

**Pull request merged**
